nemo/lightning/pytorch/strategies.py

```diff
@@ -541,7 +542,24 @@ def load_model_state_dict(self, checkpoint: Mapping[str, Any], strict: bool = Tr
     @override
     def checkpoint_io(self) -> CheckpointIO:
         if self._checkpoint_io is None:
-            self._checkpoint_io = MegatronCheckpointIO()
+            checkpoint_callback = self.trainer.checkpoint_callback
+            async_save = getattr(checkpoint_callback, "async_save", False)
+            if async_save:
+                self._checkpoint_io = AsyncFinalizableCheckpointIO(
+                    MegatronCheckpointIO(
+                        save_ckpt_format='torch_dist',
+                        async_save=True,
+                        torch_dist_multiproc=None,  ## TODO: make configurable
+                        assume_constant_structure=False,
+                        parallel_save=True,  ## TODO: make configurable
+                        parallel_load=True,  ## TODO: make configurable
+                    )
+                )
+                self.trainer.callbacks.append(AsyncFinalizerCallback())
+            else:
+                self._checkpoint_io = MegatronCheckpointIO()
```

Comment:
this should have all the parameters as the async path.
So the logic should be:
```
self._checkpoint_io = MegatronCheckpointIO(<all  the configuration>)
if async_save:
    self._checkpoint_io = AsyncFinalizableCheckpointIO(self._checkpoint_io)
    self.trainer.callbacks.append(AsyncFinalizerCallback())
```

maybe it would be worth checking if AsyncFinalizerCallback() is not already in the callbacks